lib/core/src/lib/pipes/full-name.pipe.spec.ts

```diff
@@ -16,42 +16,168 @@
+            includeEmailToken: false,
+            includeEmailParameter: false,
+            testCases: cosntBaseTestCases
+        },
+        falseIncludeEmailTokenTrueIncludeEmailParameter: {
+            title: 'and include email token is false and include email is true',
+            includeEmailToken: false,
+            includeEmailParameter: true,
+            testCases: getTestCasesWithEmailPatched()
+        },
+        trueIncludeEmailTokennUndefinedIncludeEmailParameterButEmailAddressNotPresent: {
+            title: 'and include email token is true and include email is undefined but email is not provided',
+            includeEmailToken: true,
+            includeEmailParameter: undefined,
+            testCases: cosntBaseTestCases.slice(0, cosntBaseTestCases.length - 1)
+        },
+        trueIncludeEmailTokennFalseIncludeEmailParameterButEmailAddressNotPresent: {
+            title: 'and include email token is true and include email is false but email is not provided',
+            includeEmailToken: true,
+            includeEmailParameter: false,
+            testCases: cosntBaseTestCases.slice(0, cosntBaseTestCases.length - 1)
+        },
+        trueIncludeEmailTokennTrueIncludeEmailParameterButEmailAddressNotPresent: {
+            title: 'and include email token is true and include email is true but email is not provided',
+            includeEmailToken: true,
+            includeEmailParameter: true,
+            testCases: cosntBaseTestCases.slice(0, cosntBaseTestCases.length - 1)
+        },
+        trueIncludeEmailTokenUndefinedIncludeEmailParameterAndEmailAddressPresent: {
+            title: 'and include email token is true and include email is undefined and email is provided',
+            includeEmailToken: true,
+            includeEmailParameter: undefined,
+            testCases: getTestCasesWithEmailPatched()
+        },
+        trueIncludeEmailTokenFalseIncludeEmailParameterAndEmailAddressPresent: {
+            title: 'and include email token is true and include email is false and email is provided',
+            includeEmailToken: true,
+            includeEmailParameter: false,
+            testCases: cosntBaseTestCases.slice(1, cosntBaseTestCases.length - 1).map(testCase => patchEmailAddress(testCase))
+        },
+        trueIncludeEmailTokenTrueIncludeEmailParameterAndEmailAddressPresent: {
+            title: 'and include email token is true and include email is true and email is provided',
+            includeEmailToken: true,
+            includeEmailParameter: true,
+            testCases: getTestCasesWithEmailPatched()
+        }
+    };
 
-    it('should return user eamil when firstName, lastName and username are not available', () => {
-        const user = {firstName : '', lastName : '', username: '', email: 'abcXyz@gmail.com'};
-        expect(pipe.transform(user)).toBe('abcXyz@gmail.com');
+    Object.keys(testCases).forEach(block => {
```

Comment:
this is some unconventional way of testing the pipe?

Reply:
if you followed Uncle Bob, the tests should be addressing one concern at a time, not a giant foreach having all tests in a single run

Reply:
My problem here is that if I try to test all the combinations for the injection token value and pipe parameter values, then I end up with a comment from sonar saying that I have a lot of duplicated code because the tests are always the same, call the pipe and check the output, and in many cases the output and the user are the same